HW0/P4/P4.py

```diff
@@ -30,47 +30,97 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
+    s_true=np.loadtxt('P4_trajectory.txt',delimiter=',')
+    x_coords=s_true[:,0]
+    y_coords=s_true[:,1]
+    z_coords=s_true[:,2]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    ax.plot(x_coords, y_coords, z_coords,
+             '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
-
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    measurements=np.loadtxt('P4_measurements.txt',delimiter=',')
+    C=np.zeros((3,6))
+    C[0,0]=1/rx
+    C[1,1]=1/ry
+    C[2,2]=1/rz
+    appro=np.dot(measurements,C)
+    x_coords=appro[:,0]
+    y_coords=appro[:,1]
+    z_coords=appro[:,2]
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.matrix([[1,0,0,dt,0,0], [0,1,0,0,dt,0], [0,0,1,0,0,dt], 
+        [0,0,0,1-c*dt,0,0], [0,0,0,0,1-c*dt,0], [0,0,0,0,0,1-c*dt]])
+
+    a = np.matrix([0,0,0,0,0,g*dt]).transpose()
+    s = np.zeros((6,K))
+    s0 = np.matrix([0,0,2,15,3.5,4.0])
+    s[:,0] = s0
+    s = np.asmatrix(s)
+
+    for i in xrange(0,K-1):
+        s[:,i+1]=np.add(np.dot(A,s[:,i]),a)
```

Comment:
Good point, I would have done it if it were on MATLAB. I am still working on my transition to Python and this is certainly very useful feedback!